internal/scm/github/github.go

```diff
@@ -357,6 +358,31 @@ func (g *Github) GetPullRequests(ctx context.Context, branchName string) ([]scm.
 		return nil, err
 	}
 
+	wg, ctx := errgroup.WithContext(ctx)
+	batches := chunkSlice(repos, 50)
+	var pullRequests []scm.PullRequest
+
+	for _, repos := range batches {
+		repos := repos
+		wg.Go(func() error {
```

Comment:
The GitHub API docs states

> Make requests for a single user or client ID serially. Do not make requests for a single user or client ID concurrently.

https://docs.github.com/en/rest/guides/best-practices-for-integrators?apiVersion=2022-11-28#dealing-with-secondary-rate-limits

So we should probably do that.